userspace/libsinsp/filterchecks.cpp

```diff
@@ -2782,6 +2787,174 @@ uint8_t* sinsp_filter_check_thread::extract(sinsp_evt *evt, OUT uint32_t* len, b
 				RETURN_EXTRACT_STRING(m_tstr);
 			}
 		}
+	case TYPE_VPGID_NAME:
+		{
+			bool is_host = true;
+			sinsp_threadinfo* vpgidinfo = NULL;
+			auto flags = tinfo->m_flags;
+			if((flags && flags & PPM_CL_CHILD_IN_PIDNS) || tinfo->m_tid != tinfo->m_vtid)
```

Comment:
here I would simply leave
```suggestion
			if((tinfo->m_flags & PPM_CL_CHILD_IN_PIDNS) || tinfo->m_tid != tinfo->m_vtid)
			{
			...
			}
```
and maybe we could create a useful `sinsp_threadinfo` method like `is_in_container()` that does exactly this check and return a bool. WDYT?

Reply:
With this helper we could obtain a slightly simpler code

```cpp
case TYPE_VPGID_NAME:
{
	if(!tinfo->is_in_container())
	{
		// Relying on the convention that a process group id is the process id of the process group leader.
		// `threadinfo` lookup only applies when the process is running on the host and not in a pid
		// namespace. However, if the process is running in a pid namespace, we instead traverse the process
		// lineage until we find a match.
		auto vpgidinfo = m_inspector->get_thread_ref(tinfo->m_vpgid, false).get();
		if(vpgidinfo != NULL)
		{
			m_tstr = vpgidinfo->get_comm();
			RETURN_EXTRACT_STRING(m_tstr);
		}
	}

	// This can occur when the process group leader process has exited or if the process
	// is running in a pid namespace and we only have the virtual process group id, as
	// seen from its pid namespace.
	// Find the highest ancestor process that has the same process group id and
	// declare it to be the process group leader.
	sinsp_threadinfo* group_leader = nullptr;

	int64_t vpgid = tinfo->m_vpgid;
	sinsp_threadinfo::visitor_func_t visitor = [vpgid, &group_leader](sinsp_threadinfo* pt)
	{
		if(pt->m_vpgid != vpgid)
		{
			return false;
		}
		group_leader = pt;
		return true;
	};

	tinfo->traverse_parent_state(visitor);

	// mt has been updated to the highest process that has the same process group id.
	// mt's comm is considered the process group leader.
	m_tstr = group_leader->get_comm();
	RETURN_EXTRACT_STRING(m_tstr);
}
```

Not sure it will compile, just to give you an idea of what I have in mind, moreover why do we need to get the main thread? in the end, the `vpgid` should be the same for all threads of the group

Reply:
Addressed (slightly) modified to still fall back to parent lineage traversal for host processes where leader may have exited and here we talk only about `is_in_pid_namespace` not what sinsp thinks is a container based on container engine support.

Reply:
uhm to be honest also the proposed solution :point_up: offers a fallback for the host case :thinking: I won't fight for this but if we can remove nested code (in this case an `if/else`) I would go for it